syft/file/cataloger/executable/elf.go

```diff
@@ -30,40 +31,40 @@ func findELFFeatures(data *file.Executable, reader unionreader.UnionReader) erro
 	}
 
 	data.ImportedLibraries = libs
-	data.ELFSecurityFeatures = findELFSecurityFeatures(f)
+	data.ELFSecurityFeatures = findELFSecurityFeatures(&err, f)
 	data.HasEntrypoint = elfHasEntrypoint(f)
-	data.HasExports = elfHasExports(f)
+	data.HasExports = elfHasExports(&err, f)
 
-	return nil
+	return err
 }
 
-func findELFSecurityFeatures(f *elf.File) *file.ELFSecurityFeatures {
+func findELFSecurityFeatures(errs *error, f *elf.File) *file.ELFSecurityFeatures {
```

Comment:
based off of the direction of the known-unknowns work it might be that most of these binary cases are really not known-unknown candidates, since if they were known would not result in any more nodes in the dependency tree.